src/tree.js

```diff
@@ -53,11 +53,11 @@ export function createInternal(vnode, parentInternal) {
 
 		// flags = typeof type === 'function' ? COMPONENT_NODE : ELEMENT_NODE;
 		flags =
-			type === Fragment
-				? TYPE_FRAGMENT
-				: typeof type === 'function'
+			typeof type === 'function'
 				? type.prototype && 'render' in type.prototype
 					? TYPE_CLASS
+					: props._parentDom
```

Comment:
Makes me wonder if we should add a dedicated root function in the future instead of using a Fragment. Something to try out later